File: frontend/src/objects/blog/request/CreateBlogCommentInput.ts
import type { BlogCommentId } from '@/objects/blog/BlogCommentId'
import type { BlogId } from '@/objects/blog/BlogId'
import type { CreateBlogCommentRequest } from '@/objects/blog/request/CreateBlogCommentRequest'

/** 创建博客评论的输入；顶层评论没有 parentCommentId，回复会携带父评论 ID。 */
export type CreateBlogCommentInput = {
  blogId: BlogId
  parentCommentId: BlogCommentId | null
  request: CreateBlogCommentRequest
}

File: backend/src/main/scala/domains/blog/objects/request/CreateBlogCommentInput.scala
package domains.blog.objects.request

import domains.blog.objects.{BlogCommentId, BlogId}

/** 创建博客评论的输入；顶层评论没有 parentCommentId，回复会携带父评论 ID。 */
final case class CreateBlogCommentInput(
  blogId: BlogId,
  parentCommentId: Option[BlogCommentId],
  request: CreateBlogCommentRequest
)
